pyciemss/interfaces.py

```diff
@@ -528,16 +533,204 @@ def wrapped_model():
+        model=control_model,
+        interventions=static_parameter_interventions,
+        qoi=qoi,
+        end_time=end_time,
+        logging_step_size=logging_step_size,
+        start_time=start_time,
+        risk_measure=lambda z: alpha_superquantile(z, alpha=0.95),
+        num_samples=1,
+        guide=inferred_parameters,
+        solver_method=solver_method,
+        solver_options=solver_options,
+    )
+
+    # Run one sample to estimate model evaluation time
+    start_t = time.time()
+    init_prediction = RISK.propagate_uncertainty(initial_guess_interventions)
+    RISK.qoi(init_prediction)
+    end_t = time.time()
+    forward_time = end_t - start_t
+    time_per_eval = forward_time / 1.0
+    if verbose:
+        print(f"Time taken: ({forward_time/1.:.2e} seconds per model evaluation).")
+
+    # Assign the required number of MC samples for each OUU iteration
+    RISK = computeRisk(
+        model=control_model,
+        interventions=static_parameter_interventions,
+        qoi=qoi,
+        end_time=end_time,
+        logging_step_size=logging_step_size,
+        start_time=start_time,
+        risk_measure=lambda z: alpha_superquantile(z, alpha=0.95),
+        num_samples=n_samples_ouu,
+        guide=inferred_parameters,
+        solver_method=solver_method,
+        solver_options=solver_options,
+    )
+    # Define constraints >= 0
+    constraints = (
+        # risk constraint
+        {"type": "ineq", "fun": lambda x: risk_bound - RISK(x)},
+        # bounds on control
+        {"type": "ineq", "fun": lambda x: x - u_min},
+        {"type": "ineq", "fun": lambda x: u_max - x},
+    )
+    if verbose:
+        print(
+            "Performing risk-based optimization under uncertainty (using alpha-superquantile)"
+        )
+        print(
+            f"Estimated wait time {time_per_eval*n_samples_ouu*(maxiter+1)*maxfeval:.1f} seconds..."
+        )
+    start_time = time.time()
+    opt_results = solveOUU(
+        x0=initial_guess_interventions,
+        objfun=objfun,
+        constraints=constraints,
+        maxiter=maxiter,
+        maxfeval=maxfeval,
+        u_bounds=bounds_np,
+    ).solve()
+
+    # Rounding up to given number of decimal places
+    def round_up(num, dec=roundup_decimal):
+        rnum = np.zeros(num.shape[-1])
+        for i in range(num.shape[-1]):
+            rnum[i] = ceil(num[i] * 10**dec) / (10**dec)
+        return rnum
+
+    opt_results.x = round_up(np.atleast_1d(opt_results.x))
+    if verbose:
+        print(f"Optimization completed in time {time.time()-start_time:.2f} seconds.")
+        print(f"Optimal policy:\t{opt_results.x}")
+
+    # Check for some interventions that lead to no feasible solutions
+    if opt_results.x < 0:
+        if verbose:
+            print("No solution found")
+
+    # Post-process OUU results
+    if postprocess:
+        if verbose:
+            print("Post-processing optimal policy...")
+        # TODO: change num_samples to a user-input
```

Comment:
@anirban-chaudhuri , can you address this TODO?

Reply:
I have removed this postprocess. We talked about merging parts of this to the prepare_interchange_dictionary as we had done in our previous version.